src/compositemark/callout.ts

```diff
@@ -0,0 +1,131 @@
+import {Config} from '../config';
+import {Encoding} from '../encoding';
+import * as log from '../log';
+import {MarkConfig, MarkDef} from '../mark';
+import {GenericUnitSpec, LayerSpec} from '../spec';
+
+
+export const CALLOUT: 'callout' = 'callout';
+export type CALLOUT = typeof CALLOUT;
+
+export interface CalloutDef {
+  type: CALLOUT;
+  calloutAngle?: number;
+  calloutOffset?: number;
+  calloutLength?: number;
+  labelOffset?: number;
+  callout?: MarkDef;
+  label?: MarkDef;
+}
+
+export function isCalloutDef(mark: CALLOUT | CalloutDef): mark is CalloutDef {
+  return !!mark['type'];
+}
+
+export type CalloutStyle = 'callout-label' | 'callout-rule';
+
+export const CALLOUT_STYLES: CalloutStyle[] = ['callout-label', 'callout-rule'];
+
+export interface CalloutConfig extends MarkConfig {
```

Comment:
Why does this extend mark config?  

For example, if I set "config.callout.size", would this affect the callout in some way?